opentracing/src/main/java/opentracing/TraceContextSource.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package opentracing;
+
+import java.util.Map;
+
+/**
+ * Long-lived interface that knows how to create a root {@linkplain TraceContext} and encode/decode
+ * any other.
+ */
+public interface TraceContextSource {
```

Comment:
Sounds like then the class design for `Tracer` is not taking responsibility for its actions if users of an object require access to its context. And not providing it creates the simpler API which should always be favoured. (Adding it in the future if a good use case comes along doesn't break compatibility, but removing it does).

Do we have a use-case or example where `source()` is needed?